portfolio/src/main/java/com/google/sps/servlets/BookServlet.java

```diff
@@ -50,21 +52,22 @@ public void init() throws ServletException {
           // close old book:
           if (i != 1) {
             Book book = current_builder.build();
-            listOfBooks.add(book);
+            allBooks.put(currentId,book);
+            currentId ++;
```

Comment:
nit: no space between currentId and ++.
In fact, I think you could just put this on the above line like so: allBooks.put(currentId++, book);